extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/OkHttpVaultClient.java

```diff
@@ -167,11 +174,61 @@ public VaultTransitEncrypt rewrap(String token, String keyName, VaultTransitRewr
         return post("transit/rewrap/" + keyName, token, body, VaultTransitEncrypt.class);
     }
 
+    @Override
+    public VaultTOTPCreateKeyResult createTOTPKey(String token, String keyName,
+            VaultTOTPCreateKeyBody body) {
+        String path = "totp/keys/" + keyName;
+
+        // Depending on parameters it might produce an output or not
+        if (body.isProducingOutput()) {
+            return post(path, token, body, VaultTOTPCreateKeyResult.class, 200);
+        } else {
+            post(path, token, body, 204);
+            return null;
+        }
+    }
+
+    @Override
+    public VaultTOTPReadKeyResult readTOTPKey(String token, String keyName) {
+        String path = "totp/keys/" + keyName;
+        return get(path, token, VaultTOTPReadKeyResult.class);
+    }
+
+    @Override
+    public VaultTOTPListKeysResult listTOTPKeys(String token) {
+        return list("totp/keys", token, VaultTOTPListKeysResult.class);
+    }
+
+    @Override
+    public void deleteTOTPKey(String token, String keyName) {
+        String path = "totp/keys/" + keyName;
+        delete(path, token, 204);
+    }
+
+    @Override
+    public VaultTOTPGenerateCodeResult generateTOTPCode(String token, String keyName) {
+        String path = "totp/code/" + keyName;
+        return get(path, token, VaultTOTPGenerateCodeResult.class);
+    }
+
+    @Override
+    public VaultTOTPValidateCodeResult validateTOTPCode(String token, String keyName,
+            String code) {
+        String path = "totp/code/" + keyName;
+        VaultTOTPValidateCodeBody vaultTOTPValidateCodeBody = new VaultTOTPValidateCodeBody(code);
```

Comment:
I have used `body` for local variables and parameters throughout this class for conciseness. same for `result` in `VaultTOTPManager`